extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -200,11 +208,31 @@ export class Bind {
   }
 
   /**
-   * @param {BindVsyncStateDef} unusedState
+   * Validates and removes invalid expression results from `state.results`.
+   * @param {BindVsyncStateDef} state
    * @private
    */
-  measure_(unusedState) {
-    // TODO(choumx): Validate here or in applyBinding_()?
+  measure_(state) {
+    for (let i = 0; i < this.bindings_.length; i++) {
+      const binding = this.bindings_[i];
+      const tagName = binding.element.tagName;
+      const property = binding.property;
+      const expression = binding.expression;
+      const result = state.results[expression];
+
+      // Convert result to string for validation.
+      let resultString;
+      if (property === 'class' && Array.isArray(result)) {
+        resultString = result.join(' ');
+      } else {
+        resultString = String(result);
+      }
+      if (!this.validator_.isResultValid(tagName, property, resultString)) {
```

Comment:
Design wise, can we move all references to the validator into `bind-evaluator`, so it can run on the background thread?